Test/Unit/Context.php

```diff
@@ -0,0 +1,226 @@
+                    $context->getVariable($name);
+                })
+                    ->isInstanceOf('Hoa\Math\Exception\UnknownVariable')
+                    ->hasMessage('Variable ' . $name .' does not exist');
+    }
+
+    public function case_context_returns_variable_value()
+    {
+        $this
+            ->given(
+                $name     = 'foo',
+                $value    = 42,
+                $callable = function() use ($value) { return $value; },
+                $context  = new CUT()
+            )
+            ->when($context->addVariable($name, $callable))
+            ->then
+                ->variable($context->getVariable($name))->isEqualTo($value);
+    }
+
+    public function case_context_has_predefined_constants()
+    {
+        $this
+            ->given($context = new CUT())
+            ->then
+                ->array(iterator_to_array($context->getConstants()))
+                    ->isNotEmpty()
+                    ->hasKey('PI')
+                    ->hasKey('PI_2')
+                    ->hasKey('PI_4')
+                    ->hasKey('E')
+                    ->hasKey('SQRT_PI')
+                    ->hasKey('SQRT_2')
+                    ->hasKey('SQRT_3')
+                    ->hasKey('LN_PI');
+    }
+
+    public function case_context_exception_when_getting_unknown_constant()
+    {
+        $this
+            ->given(
+                $name    = 'FOO',
+                $context = new CUT()
+            )
+            ->then
+                ->exception(function () use ($context, $name) {
+                    $context->getConstant($name);
+                })
+                    ->isInstanceOf('Hoa\Math\Exception\UnknownConstant')
+                    ->hasMessage('Constant ' . $name .' does not exist');
+    }
+
+    public function case_context_exception_when_setting_already_defined_constant()
+    {
+        $this
+            ->given(
+                $name    = 'PI',
+                $context = new CUT()
+            )
+            ->then
+                ->exception(function () use ($context, $name) {
+                    $context->addConstant($name, rand(0, PHP_INT_MAX));
+                })
+                    ->isInstanceOf('Hoa\Math\Exception\AlreadyDefinedConstant')
+                    ->hasMessage('Constant ' . $name .' is already defined');
+    }
+
+    public function case_context_returns_constant_value()
+    {
+        $this
+            ->given(
+                $name     = 'FOO',
+                $value    = 42,
+                $context  = new CUT()
+            )
+            ->when($context->addConstant($name, $value))
+            ->then
+                ->variable($context->getConstant($name))->isEqualTo($value);
+    }
+
+    public function case_context_has_predefined_functions()
+    {
+        $this
+            ->given($context = new CUT())
+            ->then
+                ->array(iterator_to_array($context->getFunctions()))
+                    ->isNotEmpty()
```

Comment:
Here I can't stricly test the contents of the array as it conains some closures.

I'll stick with `hasSize` + `hasKey`, ok ?